File: SDKs/zoom.us/2.0.0/typescript/src/sdk/models/operations/userpicture.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class UserPicturePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=userId" })
  userId: string;
}


export class UserPictureRequestBodyPicFile extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "multipart_form, content=true" })
  content: Uint8Array;

  @SpeakeasyMetadata({ data: "multipart_form, name=pic_file" })
  picFile: string;
}


export class UserPictureRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "multipart_form, file=true" })
  picFile: UserPictureRequestBodyPicFile;
}


export class UserPictureRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: UserPicturePathParams;

  @SpeakeasyMetadata({ data: "request, media_type=multipart/form-data" })
  request: UserPictureRequestBody;
}


export class UserPictureResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  headers: Record<string, string[]>;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/zoom.us/2.0.0/python/src/sdk/models/operations/userpicture.py
import dataclasses



@dataclasses.dataclass
class UserPicturePathParams:
    user_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'userId', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class UserPictureRequestBodyPicFile:
    content: bytes = dataclasses.field(metadata={'multipart_form': { 'content': True }})
    pic_file: str = dataclasses.field(metadata={'multipart_form': { 'field_name': 'pic_file' }})
    

@dataclasses.dataclass
class UserPictureRequestBody:
    pic_file: UserPictureRequestBodyPicFile = dataclasses.field(metadata={'multipart_form': { 'file': True }})
    

@dataclasses.dataclass
class UserPictureRequest:
    path_params: UserPicturePathParams = dataclasses.field()
    request: UserPictureRequestBody = dataclasses.field(metadata={'request': { 'media_type': 'multipart/form-data' }})
    

@dataclasses.dataclass
class UserPictureResponse:
    content_type: str = dataclasses.field()
    headers: dict[str, list[str]] = dataclasses.field()
    status_code: int = dataclasses.field()
    
